lib/packwerk.rb

```diff
@@ -14,7 +14,6 @@
 require "packwerk/configuration"
 require "packwerk/const_node_inspector"
 require "packwerk/constant_discovery"
-require "packwerk/constant_name_inspector"
```

Comment:
I prefer requiring a file where it is actually used because it makes the dependencies clearer. In this case, the `requires` in `lib/package.rb` would only contain the entry points into the application and what we intend to expose to our users. This also means that you will not get into a situation where you don't know what any of the `requires` is for. 

I think there is an argument for convenience by requiring every file in the `lib/package.rb` file, but I think having clearly defined `requires` makes the codebase easier to understand in the long run.

I would like us to align one approach since this allows us to be consistent in the codebase, but I'm happy to go with either approach. So thank you for raising this issue :)